pkgs/top-level/gnuradio-packages.nix

```diff
@@ -37,7 +38,9 @@ in {
   inherit gnuradio;
   inherit (gnuradio) python;
 
-  osmosdr = callPackage ../development/gnuradio-modules/osmosdr/default.nix { };
+  osmosdr = callPackage ../development/gnuradio-modules/osmosdr/default.nix {
+    inherit darwin;
```

Comment:
Are you sure that's needed?

Reply:
It is required to fix the following error:

```
       last 10 log lines:
       > 55 warnings generated.
       > 56 warnings generated.
       > 53 warnings generated.
       > 48 warnings generated.
       > [ 90%] Linking CXX shared library libgnuradio-osmosdr.dylib
       > ld: file not found: /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit for architecture arm64
       > clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
       > make[2]: *** [lib/CMakeFiles/gnuradio-osmosdr.dir/build.make:508: lib/libgnuradio-osmosdr.0.2.0.0.dylib] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:418: lib/CMakeFiles/gnuradio-osmosdr.dir/all] Error 2
       > make: *** [Makefile:146: all] Error 2
       For full logs, run 'nix log /nix/store/a6h8zf69cxzbdzv89y30i7gf5hs0p8ki-gr-osmosdr-0.2.3.drv'.
```

Reply:
> It is required to fix the following error:

I understand that IOKit is needed, and I didn't mean to suggest to remove it. What I meant is that `callPackage` and also the `callPackage` that is created by `newScope` should know what is `darwin` and send that argument to `osmosdr`. I can't test it myself because I can't force nixpkgs to be evaluated as if I have a darwin system, but I was thinking about something like this (on top of current changes):

```diff
diff --git i/pkgs/applications/radio/gnuradio/wrapper.nix w/pkgs/applications/radio/gnuradio/wrapper.nix
index 89df489f70e..926022353e4 100644
--- i/pkgs/applications/radio/gnuradio/wrapper.nix
+++ w/pkgs/applications/radio/gnuradio/wrapper.nix
@@ -24,7 +24,6 @@
 , harfbuzz
 , at-spi2-core
 , atk
-, darwin
 # For Adding additional GRC blocks
 , extraPackages ? []
 # For Adding additional python packaages
@@ -126,7 +125,7 @@ let
   );
 
   packages = import ../../../top-level/gnuradio-packages.nix {
-    inherit lib stdenv newScope darwin;
+    inherit lib stdenv newScope;
     gnuradio = unwrapped;
   };
   passthru = unwrapped.passthru // {
diff --git i/pkgs/top-level/gnuradio-packages.nix w/pkgs/top-level/gnuradio-packages.nix
index b23771059bb..1cd2c93da77 100644
--- i/pkgs/top-level/gnuradio-packages.nix
+++ w/pkgs/top-level/gnuradio-packages.nix
@@ -2,7 +2,6 @@
 , stdenv
 , newScope
 , gnuradio # unwrapped gnuradio
-, darwin
 }:
 
 lib.makeScope newScope ( self:
@@ -38,9 +37,7 @@ in {
   inherit gnuradio;
   inherit (gnuradio) python;
 
-  osmosdr = callPackage ../development/gnuradio-modules/osmosdr/default.nix {
-    inherit darwin;
-  };
+  osmosdr = callPackage ../development/gnuradio-modules/osmosdr/default.nix { };
 
   ais = callPackage ../development/gnuradio-modules/ais/default.nix { };
 
```

Reply:
@jiegec could you please check if commit `gnuradio3_8Packages.osmosdr: add missing framework on darwin` needs all of the `inherit darwin;` lines?

Reply:
Yes, you are right, the lines are not needed.